server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -177,6 +178,76 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         );
     }
 
+    /**
+     * Removes the provided legacy index templates if they exist. This only matches the templates using exact name, wildcards are not
+     * supported.
+     *
+     * The provided templates must exist in the cluster, otherwise an {@link IndexTemplateMissingException} is reported.
+     */
+    public static void removeTemplates(
```

Comment:
maybe rename to `removeLegacyTemplates`? To emphasise this batch removal method is for legacy templates only.